### PR TITLE
Removed the cloneOperator and replaced with kryo

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

File: contrib/src/test/java/com/datatorrent/contrib/kafka/SimpleKakfaConsumerTest.java [hunk 1]
```diff
@@ -23,8 +23,13 @@ public void cloneTest()
     kc.setTopic("test_topic");
     kc.setClientId("test_clientid");
 
-    @SuppressWarnings({ "unchecked", "rawtypes" })
-    SimpleKafkaConsumer kcClone = (SimpleKafkaConsumer) kc.cloneConsumer(new HashSet(), new HashMap());
+    Kryo kryo = new Kryo();
```
Comment (tweise) [inline contrib/src/test/java/com/datatorrent/contrib/kafka/SimpleKakfaConsumerTest.java]: Use TestUtil.clone().

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractPartitionableKafkaInputOperator.java [hunk 2]
```diff
@@ -311,6 +280,25 @@ else if (newWaitingPartition.size() != 0) {
     return newPartitions;
   }
 
+  // Create a new partition with the shardIds and initial shard positions
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractPartitionableKafkaInputOperator.java]: Was this copied from Kinesis?

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractPartitionableKafkaInputOperator.java [hunk 3]
```diff
@@ -311,6 +280,25 @@ else if (newWaitingPartition.size() != 0) {
...
+  private
+  Partition<AbstractPartitionableKafkaInputOperator> createPartition(Set<KafkaPartition> pIds, Map<KafkaPartition, Long> initOffsets)
+  {
+    Kryo kryo = new Kryo();
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractPartitionableKafkaInputOperator.java]: The clone part should be in a protected method so derived class has a chance to override it.

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractPartitionableKafkaInputOperator.java (hunk 2 above)
Comment (chaithu14) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractPartitionableKafkaInputOperator.java]: By-mistake, I copied the comment.

**Comment by chaithu14** [comment]
Updated the code as per the changes.

**Pull request merged**